File: 0953-verifying-an-alien-dictionary/0953-verifying-an-alien-dictionary.py
class Solution:
    def isAlienSorted(self, words: List[str], order: str) -> bool:
        mp = { ch : indx for indx,ch in enumerate(order)}

        for i in range(len(words)-1):
            fword,sword = words[i],words[i+1]
            for j in range(len(fword)):
                if j == len(sword): return False
                if fword[j] != sword[j]:
                    if mp[fword[j]] > mp[sword[j]]: return False
                    break
        return True
        

File: 0953-verifying-an-alien-dictionary/0953-verifying-an-alien-dictionary.cpp
class Solution {
public:
    bool isAlienSorted(vector<string>& words, string order) {
        unordered_map<char,int> mp;
        for(int i = 0 ; i < order.size() ; i++){
            mp[order[i]] = i ;
        }
        for(int i = 0 ; i < words.size()-1; i++){
            string fword = words[i];
            string sword = words[i+1];
            for(int j = 0 ; j < fword.size(); j++){
                // if the first word is a prefix of the second one we return false
                if(j == sword.size()) return false;
                if(fword[j] != sword[j]){
                    if(mp[fword[j]] > mp[sword[j]]) return false;
                    break;
                }
                
            }
        }
        return true;

        
    }
};